diagnosis/getkuberneteslogs.sh

```diff
@@ -128,56 +121,89 @@ done
 
 if [ -z "$IDENTITYFILE" ]
 then
     echo ""
-    echo "[ERR] --identity-file is required"
+    echo -e "$(date) [Err] --identity-file is required"
     printUsage
 fi
 
-if [ -z "$DVM_HOST" -a -z "$MASTER_HOST" ]
+if [ -z "$DVM_HOST" ] || [ -z "$MASTER_HOST" ]
 then
     echo ""
-    echo "[ERR] Either --vmd-host or --master-host should be provided"
-    printUsage
+    echo -e "$(date) [Err] --vmd-host or --master-host is required"
 fi
 
 if [ ! -f $IDENTITYFILE ]
 then
     echo ""
-    echo "[ERR] identity-file not found at $IDENTITYFILE"
+    echo -e "$(date) [Err] identity-file not found at $IDENTITYFILE"
     printUsage
     exit 1
 else
     cat $IDENTITYFILE | grep -q "BEGIN \(RSA\|OPENSSH\) PRIVATE KEY" \
-    || { echo "The identity file $IDENTITYFILE is not a RSA Private Key file."; echo "A RSA private key file starts with '-----BEGIN [RSA|OPENSSH] PRIVATE KEY-----''"; exit 1; }
+    || { echo -e "$(date) [Err] The identity file $IDENTITYFILE is not a RSA Private Key file."; log_level -e "A RSA private key file starts with '-----BEGIN [RSA|OPENSSH] PRIVATE KEY-----''"; exit 1; }
 fi
 
-test $ALLNAMESPACES -eq 0 && unset NAMESPACES
-
-# Print user input
-echo ""
-echo "user:             $USER"
-echo "identity-file:    $IDENTITYFILE"
-echo "master-host:      $MASTER_HOST"
-echo "vmd-host:         $DVM_HOST"
-echo "namespaces:       ${NAMESPACES:-all}"
-echo ""
+if [[ $ALLNAMESPACES -eq 0 ]];then
+    unset NAMESPACES
+    NAMESPACES="all"
+fi
 
 NOW=`date +%Y%m%d%H%M%S`
 CURRENTDATE=$(date +"%Y-%m-%d-%H-%M-%S-%3N")
-LOGFILEFOLDER="./KubernetesLogs_$CURRENTDATE"
-SCRIPTSFOLDER="$LOGFILEFOLDER/scripts"
-mkdir -p $LOGFILEFOLDER/scripts
+LOGFILEFOLDER="./Diagnosis/KubernetesLogs_$CURRENTDATE"
+SCRIPTS_FOLDER="./Diagnosis/Scripts"
+mkdir -p $SCRIPTS_FOLDER
+mkdir -p $LOGFILEFOLDER
 mkdir -p ~/.ssh
 
+ARTIFACTSURL="${ARTIFACTSURL:-https://raw.githubusercontent.com/msazurestackworkloads/azurestack-gallery/2.0}"
+
+#Loading defaults
+if [ -f ./defaults.sh ]; then
+    echo -e "$(date) [Info] Using local default settings"
+    source ./defaults.sh
+else
+    echo -e "$(date) [Info] Downloading Defaults"
+    curl -fs $ARTIFACTSURL/diagnosis/defaults.sh -o ./defaults.sh
+    source ./defaults.sh
+fi
+
 # Download scripts from github
-ARTIFACTSURL="${ARTIFACTSURL:-https://raw.githubusercontent.com/msazurestackworkloads/azurestack-gallery/master}"
-download_scripts $ARTIFACTSURL
+if [[ $FORCE_DOWNLOAD == "yes" ]]; then
+    echo -e "$(date) [Info] Downloading scripts and overwriting"
+    download_scripts $ARTIFACTSURL
+else
+    if [[ -z "$(ls -A $SCRIPTS_FOLDER)" ]]; then
+        echo -e "$(date) [Info] Scripts not available locally downloading"
+        download_scripts $ARTIFACTSURL
+    else
+        echo -e "$(date) [Info] Scripts available locally... skipping"
+    fi
+fi
+
+source $SCRIPTS_FOLDER/common.sh $LOGFILEFOLDER "getkuberneteslogs"
+
+log_level -i "-----------------------------------------------------------------------------"
```

Comment:
are these scripts what we will ask users to run in order to collect logs?

In that case, then we should list the param names as printed by `printUsage`.
The new `RUN_***` parameters may confuse users, so far it looks like there is nothing that talks about them

Reply:
These variables cannot be instantiated in the command line. They were purposely left out of the command line arguments as they would increase the number of  configuration options the user would have to remember. You do raise a good point and as such I would include comments in the defaults file to detail what each option does when set to true.